modules/apps/remote-app/remote-app-service/src/main/java/com/liferay/remote/app/internal/upgrade/v2_4_0/RemoteAppEntryUpgradeProcess.java

```diff
@@ -0,0 +1,56 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.remote.app.internal.upgrade.v2_4_0;
+
+import com.liferay.petra.string.StringBundler;
+import com.liferay.portal.kernel.dao.db.DBTypeToSQLMap;
+import com.liferay.portal.kernel.upgrade.UpgradeProcess;
+import com.liferay.remote.app.internal.upgrade.v2_4_0.util.RemoteAppEntryTable;
+
+/**
+ * @author José Abelenda
+ */
+public class RemoteAppEntryUpgradeProcess extends UpgradeProcess {
+
+	protected void addExternalReferenceCodeColumn() throws Exception {
+		if (!hasColumn(
+				RemoteAppEntryTable.TABLE_NAME, "externalReferenceCode")) {
+
+			alter(
+				RemoteAppEntryTable.class,
+				new AlterTableAddColumn(
+					"externalReferenceCode", "VARCHAR(75)"));
+		}
+	}
+
+	@Override
+	protected void doUpgrade() throws Exception {
+		addExternalReferenceCodeColumn();
+
+		updateExternalReferenceCode();
+	}
+
+	protected void updateExternalReferenceCode() throws Exception {
+		StringBundler sb = new StringBundler(2);
+
+		sb.append("update RemoteAppEntry set externalReferenceCode = ");
+		sb.append("remoteAppEntryId where externalReferenceCode is null ");
```

Comment:
Don't know if this SQL Query will work considering that the remoteAppEntryId is a bigint and the externalReferenceCode a VarChar.

In other upgrade processes we used the CAST_TEXT function to convert it

Reply:
Yes, it works. I tested before send PR.
But I preferred follow your suggestion and use CAST_TEXT to make it consistent with the pattern.